org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/views/issues/ChangeSetIssuesView.java

```diff
@@ -0,0 +1,141 @@
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+public class ChangeSetIssuesView extends MarkerSupportView {
+
+  public static final String ID = SonarLintUiPlugin.PLUGIN_ID + ".views.issues.ChangeSetIssuesView";
+  private static LocalDateTime lastRefresh;
+  private static ChangeSetIssuesView instance;
+  private Label label;
+  private Composite bottom;
+  private Button btn;
+
+  public ChangeSetIssuesView() {
+    super(SonarLintUiPlugin.PLUGIN_ID + ".views.issues.changeSetIssueMarkerGenerator");
+    instance = this;
+  }
+
+  @Override
+  public void createPartControl(Composite parent) {
+    GridLayout layout = new GridLayout(1, false);
+    layout.marginHeight = 0;
+    layout.marginWidth = 0;
+    layout.horizontalSpacing = 0;
+    layout.verticalSpacing = 0;
+    parent.setLayout(layout);
+    Composite issuesTable = new Composite(parent, SWT.NONE);
+    GridData issuesLayoutData = new GridData(SWT.FILL, SWT.FILL, true, true);
+    issuesTable.setLayoutData(issuesLayoutData);
+    super.createPartControl(issuesTable);
+    bottom = new Composite(parent, SWT.NONE);
+    RowLayout bottomLayout = new RowLayout();
+    bottomLayout.center = true;
+    bottom.setLayout(bottomLayout);
+    GridData bottomLayoutData = new GridData(SWT.FILL, SWT.FILL, true, false);
+    bottom.setLayoutData(bottomLayoutData);
+    btn = new Button(bottom, SWT.PUSH);
+    btn.setImage(SonarLintImages.RUN_IMG);
+    btn.setText("Analyze changed files");
+    btn.addListener(SWT.Selection, e -> {
+      triggerAnalysis(asList(ResourcesPlugin.getWorkspace().getRoot().getProjects()).stream()
+        .filter(IProject::isAccessible)
+        .collect(toList()));
+    });
+    label = new Label(bottom, SWT.NONE);
+    refreshText();
+  }
+
+  private void refreshText() {
+    String msg = "Find issues on changed files based on the SCM";
+    if (lastRefresh != null) {
+      label.setText(msg + ". Report generated on " + DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm").format(lastRefresh));
+    } else {
+      label.setText(msg);
+    }
+  }
+
+  public static void setRefreshTime(LocalDateTime now) {
+    ChangeSetIssuesView.lastRefresh = now;
+    instance.refreshText();
+    instance.bottom.requestLayout();
+  }
+
+  public static void triggerAnalysis(Collection<IProject> selectedProjects) {
+    // Disable button
+    ChangeSetIssuesView.instance.btn.setEnabled(false);
+    AnalyzeChangedFilesJob job = new AnalyzeChangedFilesJob(selectedProjects);
+    registerJobListener(job);
+    job.schedule();
+  }
+
+  private static void registerJobListener(Job job) {
+    job.addJobChangeListener(new JobChangeAdapter() {
+      @Override
+      public void done(IJobChangeEvent event) {
+        Display.getDefault().asyncExec(() -> {
+          // Enable button
+          ChangeSetIssuesView.instance.btn.setEnabled(true);
+          if (Status.OK_STATUS == event.getResult()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)